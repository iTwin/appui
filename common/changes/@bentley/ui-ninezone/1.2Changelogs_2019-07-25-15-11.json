{
  "changes": [
    {
      "comment": "",
      "packageName": "@bentley/ui-ninezone",
      "type": "none"
    }
  ],
  "packageName": "@bentley/ui-ninezone",
  "email": "31107829+calebmshafer@users.noreply.github.com"
}