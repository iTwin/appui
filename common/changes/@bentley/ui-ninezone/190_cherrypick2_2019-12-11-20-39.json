{
  "changes": [
    {
      "packageName": "@bentley/ui-ninezone",
      "comment": "",
      "type": "none"
    }
  ],
  "packageName": "@bentley/ui-ninezone",
  "email": "Colin.Kerr@bentley.com"
}