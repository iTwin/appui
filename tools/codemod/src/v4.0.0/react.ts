/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
import type { API, FileInfo, Options } from "jscodeshift";
import { useExtensions } from "../utils/Extensions";

export default function transformer(
  file: FileInfo,
  api: API,
  options: Options
) {
  const j = api.jscodeshift;
  useExtensions(j);

  const root = j(file.source);

  root.rename(
    "@itwin/appui-react:CommonWidgetProps",
    "@itwin/appui-react:Widget"
  );
  root.rename("@itwin/appui-react:WidgetProps", "@itwin/appui-react:Widget");

  return root.toSource(options.printOptions);
}
