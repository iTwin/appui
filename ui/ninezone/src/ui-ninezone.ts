/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
export * from "./ui-ninezone/backstage/Backstage";
export * from "./ui-ninezone/backstage/Item";
export * from "./ui-ninezone/backstage/Separator";
export * from "./ui-ninezone/backstage/UserProfile";

export * from "./ui-ninezone/base/WithContainIn";

export * from "./ui-ninezone/footer/dialog/Button";
export * from "./ui-ninezone/footer/dialog/Dialog";
export * from "./ui-ninezone/footer/dialog/TitleBar";

export * from "./ui-ninezone/footer/message/Button";
export * from "./ui-ninezone/footer/message/Hyperlink";
export * from "./ui-ninezone/footer/message/Layout";
export * from "./ui-ninezone/footer/message/Message";
export * from "./ui-ninezone/footer/message/Progress";
export * from "./ui-ninezone/footer/message/Status";
export * from "./ui-ninezone/footer/message/Toast";

export * from "./ui-ninezone/footer/message-center/Dialog";
export * from "./ui-ninezone/footer/message-center/Indicator";
export * from "./ui-ninezone/footer/message-center/Message";
export * from "./ui-ninezone/footer/message-center/Tab";

export * from "./ui-ninezone/footer/snap-mode/Indicator";
export * from "./ui-ninezone/footer/snap-mode/Panel";
export * from "./ui-ninezone/footer/snap-mode/Snap";

export * from "./ui-ninezone/footer/tool-assistance/Dialog";
export * from "./ui-ninezone/footer/tool-assistance/Indicator";
export * from "./ui-ninezone/footer/tool-assistance/Item";
export * from "./ui-ninezone/footer/tool-assistance/Separator";

export * from "./ui-ninezone/footer/Footer";
export * from "./ui-ninezone/footer/Indicator";
export * from "./ui-ninezone/footer/Popup";
export * from "./ui-ninezone/footer/Separator";

export * from "./ui-ninezone/popup/Tooltip";

export * from "./ui-ninezone/toolbar/item/Item";
export * from "./ui-ninezone/toolbar/item/Overflow";
export * from "./ui-ninezone/toolbar/item/expandable/Expandable";

export * from "./ui-ninezone/toolbar/item/expandable/group/BackArrow";
export * from "./ui-ninezone/toolbar/item/expandable/group/Column";
export * from "./ui-ninezone/toolbar/item/expandable/group/Columns";
export * from "./ui-ninezone/toolbar/item/expandable/group/Group";
export * from "./ui-ninezone/toolbar/item/expandable/group/Nested";
export * from "./ui-ninezone/toolbar/item/expandable/group/Panel";
export * from "./ui-ninezone/toolbar/item/expandable/group/Title";

export * from "./ui-ninezone/toolbar/item/expandable/group/tool/Expander";
export * from "./ui-ninezone/toolbar/item/expandable/group/tool/Tool";

export * from "./ui-ninezone/toolbar/item/expandable/history/Icon";
export * from "./ui-ninezone/toolbar/item/expandable/history/Item";
export * from "./ui-ninezone/toolbar/item/expandable/history/Tray";

export * from "./ui-ninezone/toolbar/scroll/Chevron";
export * from "./ui-ninezone/toolbar/scroll/Indicator";

export * from "./ui-ninezone/toolbar/Items";
export * from "./ui-ninezone/toolbar/Scrollable";
export * from "./ui-ninezone/toolbar/Toolbar";

export * from "./ui-ninezone/utilities/Cell";
export * from "./ui-ninezone/utilities/Css";
export * from "./ui-ninezone/utilities/Direction";
export * from "./ui-ninezone/utilities/Point";
export * from "./ui-ninezone/utilities/Rectangle";
export * from "./ui-ninezone/utilities/Size";

export * from "./ui-ninezone/widget/Stacked";
export * from "./ui-ninezone/widget/Tools";
export * from "./ui-ninezone/widget/ToolSettings";

export * from "./ui-ninezone/widget/tools/button/App";
export * from "./ui-ninezone/widget/tools/button/Back";
export * from "./ui-ninezone/widget/tools/button/Button";
export * from "./ui-ninezone/widget/tools/button/Expandable";
export * from "./ui-ninezone/widget/tools/button/Icon";

export * from "./ui-ninezone/widget/rectangular/Content";
export * from "./ui-ninezone/widget/rectangular/ResizeGrip";
export * from "./ui-ninezone/widget/rectangular/ResizeHandle";

export * from "./ui-ninezone/widget/rectangular/tab/Group";
export * from "./ui-ninezone/widget/rectangular/tab/Separator";
export * from "./ui-ninezone/widget/rectangular/tab/Tab";

export * from "./ui-ninezone/widget/tool-settings/Nested";
export * from "./ui-ninezone/widget/tool-settings/Popup";
export * from "./ui-ninezone/widget/tool-settings/Scrollable";
export * from "./ui-ninezone/widget/tool-settings/Tab";

export * from "./ui-ninezone/zones/Outline";
export * from "./ui-ninezone/zones/Zone";
export * from "./ui-ninezone/zones/Zones";

export * from "./ui-ninezone/zones/state/Manager";
export * from "./ui-ninezone/zones/state/NineZone";
export * from "./ui-ninezone/zones/state/Target";
export * from "./ui-ninezone/zones/state/Widget";
export * from "./ui-ninezone/zones/state/Zone";

export * from "./ui-ninezone/zones/state/layout/Layout";
export * from "./ui-ninezone/zones/state/layout/Layouts";
export * from "./ui-ninezone/zones/state/layout/Root";

export * from "./ui-ninezone/zones/target/Back";
export * from "./ui-ninezone/zones/target/Merge";

// Set the version number so it can be found at runtime. BUILD_SEMVER is replaced at build time by the webpack DefinePlugin.
declare var BUILD_SEMVER: string;
/* istanbul ignore next */
if ((typeof (BUILD_SEMVER) !== "undefined") && (typeof window !== "undefined") && window) {
  if (!(window as any).iModelJsVersions)
    (window as any).iModelJsVersions = new Map<string, string>();
  (window as any).iModelJsVersions.set("ui-ninezone", BUILD_SEMVER);
}

/** @docs-package-description
 * The ui-ninezone package contains React components for application user interface layouts following the Bentley 9-Zone pattern.
 * For more information, see [learning about ui-ninezone]($docs/learning/ninezone/index.md).
 */
/**
 * @docs-group-description App
 * Classes for working with a App
 */
/**
 * @docs-group-description Backstage
 * Classes for working with a Backstage
 */
/**
 * @docs-group-description Base
 * Base classes for for working with the 9-Zone pattern
 */
/**
 * @docs-group-description Button
 * Classes for working with a Button
 */
/**
 * @docs-group-description Footer
 * Classes for working with a Footer or Status Bar
 */
/**
 * @docs-group-description Message
 * Classes for working with Messages
 */
/**
 * @docs-group-description MessageCenter
 * Classes for working with the MessageCenter
 */
/**
 * @docs-group-description Popup
 * Classes for working with a Popup
 */
/**
 * @docs-group-description SnapMode
 * Classes for working with SnapMode UI
 */
/**
 * @docs-group-description Theme
 * Provides ability to switch between different 9-Zone styles
 */
/**
 * @docs-group-description ToolAssistance
 * Classes for working with Tool Assistance
 */
/**
 * @docs-group-description Toolbar
 * Classes for working with a Toolbar
 */
/**
 * @docs-group-description ToolSettings
 * Classes for working with Tool Settings
 */
/**
 * @docs-group-description Utilities
 * Utility classes for working with the 9-Zone pattern
 */
/**
 * @docs-group-description Widget
 * Classes for working a Widget
 */
/**
 * @docs-group-description Zone
 * Classes for working a Zone
 */
