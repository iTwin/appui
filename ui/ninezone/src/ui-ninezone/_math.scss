/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/

// https://en.wikipedia.org/wiki/Newton%27s_method#Square_root_of_a_number
@function sqrt($x, $iterations: 10) {
  $sqr: 1;

  @for $i from 1 through $iterations {
    $sqr: $sqr - ($sqr * $sqr - $x) / (2 * $sqr);
  }

  @return $sqr;
}