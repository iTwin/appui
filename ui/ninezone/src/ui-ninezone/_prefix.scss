/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@mixin nz-backface-visibility($value) {
  -webkit-backface-visibility: $value;
  -moz-backface-visibility: $value;
  backface-visibility: $value;
}

@mixin nz-transition($value) {
  -webkit-transition: $value;
  -moz-transition: $value;
  -o-transition: $value;
  transition: $value;
}

@mixin nz-transform($value) {
  -moz-transform: $value;
  -webkit-transform: $value;
  transform: $value;
}

@mixin nz-border-radius($value) {
  -webkit-border-radius: $value;
  -moz-border-radius: $value;
  border-radius: $value;
}
