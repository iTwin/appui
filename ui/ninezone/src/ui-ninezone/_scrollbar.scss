/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "@bentley/ui-core/lib/ui-core/_scrollbar";
@import "@bentley/ui-core/lib/ui-core/style/themecolors";

$nz-scrollbar-width: 12px;
$nz-scrollbar-padding: 2px;

@mixin nz-scrollbar($color: $buic-background-scrollbar, $width: $nz-scrollbar-width, $padding: $nz-scrollbar-padding) {
  @include uicore-scrollbar($color, $width, $padding);
}

@mixin nz-hidden-scrollbar {
  @include uicore-hidden-scrollbar;
}
