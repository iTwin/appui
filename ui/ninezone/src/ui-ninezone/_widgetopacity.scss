/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "@bentley/ui-core/lib/ui-core/style/themecolors";

@mixin nz-widget-opacity {
  opacity: $buic-widget-opacity;

  &:hover {
    opacity: 1.00;
  }
}
