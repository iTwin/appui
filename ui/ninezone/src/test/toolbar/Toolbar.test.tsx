/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import * as React from "react";
import { mount, shallow } from "enzyme";
import { Direction, Toolbar, ToolbarPanelAlignment, PanelsProvider } from "../../ui-ninezone";

describe("<Toolbar />", () => {
  it("should render", () => {
    mount(<Toolbar />);
  });

  it("renders correctly", () => {
    shallow(<Toolbar />).should.matchSnapshot();
  });

  it("renders with expandsTo", () => {
    const sut = shallow(
      <Toolbar
        expandsTo={Direction.Right}
      />,
    );
    const renderProp = sut.find(PanelsProvider).prop("children");
    const rendered = shallow(renderProp!(undefined) as React.ReactElement<{}>);
    rendered.should.matchSnapshot();
  });

  it("renders with panelAlignment", () => {
    const sut = shallow(
      <Toolbar
        panelAlignment={ToolbarPanelAlignment.End}
      />,
    );
    const renderProp = sut.find(PanelsProvider).prop("children");
    const rendered = shallow(renderProp!(undefined) as React.ReactElement<{}>);
    rendered.should.matchSnapshot();
  });
});
