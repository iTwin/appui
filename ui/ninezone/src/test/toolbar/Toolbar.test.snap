// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> renders correctly 1`] = `
<PanelsProvider
  histories={null}
  panels={null}
>
  <Component />
</PanelsProvider>
`;

exports[`<Toolbar /> renders with expandsTo 1`] = `
<div
  className="nz-toolbar-toolbar nz-direction-right nz-vertical nz-panel-alignment-start"
>
  <div
    className="nz-expanded nz-histories"
  />
  <div
    className="nz-expanded nz-panels"
  />
  <Items
    className="nz-items"
    direction={0}
  />
</div>
`;

exports[`<Toolbar /> renders with panelAlignment 1`] = `
<div
  className="nz-toolbar-toolbar nz-direction-bottom nz-horizontal nz-panel-alignment-end"
>
  <div
    className="nz-expanded nz-histories"
  />
  <div
    className="nz-expanded nz-panels"
  />
  <Items
    className="nz-items"
    direction={1}
  />
</div>
`;
